src/main/java/io/stargate/sgv2/jsonapi/exception/playing/ErrorCode.java

```diff
@@ -0,0 +1,75 @@
+package io.stargate.sgv2.jsonapi.exception.playing;
+
+import com.google.common.base.Preconditions;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Interface for any enum that represents an error code to implement.
+ *
+ * <p>This interface is used because multiple ENUM's define the scopes, the interface creates a way
+ * to treat the values from the different ENUM's in a consistent way.
+ *
+ * <p>The interface makes it easy for code to get an instance of the exception the code represents,
+ * built using the templates error information in {@link ErrorTemplate} that is loaded at startup.
+ *
+ * <p>With this interface code creates an instance of the exception by calling any of the {@link
+ * #get()} overloads for example:
+ *
+ * <pre>
+ *   FilterException.Code.MULTIPLE_ID_FILTER.get("variable_name", "value");
+ *   RequestException.Code.FAKE_CODE.get();
+ * </pre>
+ *
+ * @param <T> Type of the {@link APIException} the error code creates.
+ */
+public interface ErrorCode<T extends APIException> {
+
+  /**
+   * Gets an instance of the {@link APIException} the error code represents without providing any
+   * substitution values for the error body.
+   *
+   * @return Instance of {@link APIException} the error code represents.
+   */
+  default T get() {
```

Comment:
This is not a good method name. Get what?

So suggesting we rename to something like "toApiException()" or "asApiException()"

Reply:
will look for better name - discussed